triton-vm/src/proof_item.rs

```diff
@@ -145,65 +319,252 @@ impl ProofItem {
+    fn encode(&self) -> Vec<BFieldElement> {
+        let mut tail = match self {
+            ProofItem::CompressedAuthenticationPaths(something) => something.encode(),
+            ProofItem::TransposedBaseElementVectors(something) => something.encode(),
+            ProofItem::TransposedExtensionElementVectors(something) => something.encode(),
+            ProofItem::MerkleRoot(something) => something.encode(),
+            ProofItem::TransposedBaseElements(something) => something.encode(),
+            ProofItem::TransposedExtensionElements(something) => something.encode(),
+            ProofItem::AuthenticationPath(something) => something.encode(),
+            ProofItem::RevealedCombinationElement(something) => something.encode(),
+            ProofItem::RevealedCombinationElements(something) => something.encode(),
+            ProofItem::FriCodeword(something) => something.encode(),
+            ProofItem::FriResponse(something) => something.encode(),
+            ProofItem::PaddedHeight(something) => something.encode(),
+            ProofItem::Uncast(something) => something.encode(),
+        };
+        let head = BFieldElement::new(tail.len().try_into().unwrap());
+        tail.insert(0, head);
+        tail
+    }
+}
 
-                bs.into_iter()
-            }
-            ProofItem::CompressedAuthenticationPaths(partial_auth_paths) => {
-                let mut bs: Vec<BFieldElement> = vec![];
+#[cfg(test)]
+mod proof_item_typed_tests {
+    use itertools::Itertools;
+    use rand::{thread_rng, RngCore};
 
-                for partial_auth_path in partial_auth_paths.iter() {
-                    for bs_in_partial_auth_path in partial_auth_path.0.iter().flatten() {
-                        bs.append(&mut bs_in_partial_auth_path.to_sequence());
-                    }
-                }
+    use crate::proof_stream::ProofStream;
 
-                bs.into_iter()
-            }
-            ProofItem::TransposedBaseElementVectors(bss) => bss.concat().into_iter(),
-            ProofItem::TransposedExtensionElementVectors(xss) => {
-                xss.into_iter().map(|xs| xs_to_bs(&xs)).concat().into_iter()
-            }
-            ProofItem::PaddedHeight(padded_height) => vec![padded_height].into_iter(),
+    use super::*;
+    use twenty_first::shared_math::{
+        b_field_element::BFieldElement, rescue_prime_regular::RescuePrimeRegular,
+        x_field_element::XFieldElement,
+    };
+
+    fn random_bool() -> bool {
+        let mut rng = thread_rng();
+        rng.next_u32() % 2 == 1
+    }
+
+    fn random_bfieldelement() -> BFieldElement {
+        let mut rng = thread_rng();
+        BFieldElement::new(rng.next_u64())
+    }
+
+    fn random_xfieldelement() -> XFieldElement {
+        XFieldElement {
+            coefficients: [
+                random_bfieldelement(),
+                random_bfieldelement(),
+                random_bfieldelement(),
+            ],
```

Comment:
Use the `random_elements` function from `twenty-first` instead of making new functions.